crates/modelardb_server/src/storage/compressed_data_manager.rs

```diff
@@ -350,11 +368,29 @@ impl CompressedDataManager {
         // Pass the saved compressed file to the data transfer component if a remote data folder
         // was provided. If the total size of the files related to table_name have reached the
         // transfer threshold, the files are transferred to the remote object store.
-        if let Some(data_transfer) = &mut self.data_transfer {
+        if let Some(data_transfer) = &*self.data_transfer.read().await {
             data_transfer
                 .add_compressed_file(table_name, column_index, file_path.as_path())
                 .await
-                .map_err(|error| IOError::new(Other, error.to_string()))?;
+                .map_err(|error| IOError::new(ErrorKind::Other, error.to_string()))?;
+        }
+
+        // Update the remaining memory for compressed data and record the change.
```

Comment:
Maybe remove `and record the change` since that is done above.

Reply:
Fixed.